projects/packages/autoloader/tests/php/bin/test-coverage.php

```diff
@@ -0,0 +1,247 @@
+		exit( -1 );
+	}
+
+	return $argv[1];
+}
+
+/**
+ * Attempts to load the report from the tmp file we should have generated.
+ *
+ * @return SebastianBergmann\CodeCoverage\CodeCoverage The unserialized code coverage object.
+ */
+function load_report() {
+	$coverage_report = implode( DIRECTORY_SEPARATOR, array( ROOT_DIR, 'tests', 'php', 'tmp', 'coverage-report.php' ) );
+	if ( ! file_exists( $coverage_report ) ) {
+		echo "There is no coverage report to process.\n";
+		exit( -1 );
+	}
+
+	return require_once $coverage_report;
+}
+
+/**
+ * Evaluates the version of sebastianbergmann/php-code-coverage that we've generated the coverage report using.
+ *
+ * @return string The version for the code coverage package.
+ */
+function get_coverage_version() {
+	return Version::id();
+}
+
+/**
+ * Counts the number of lines in a file before the `class` keyword.
+ *
+ * @param string $file The file to check.
+ * @return int|null The number of lines or null if there is no class keyword.
+ */
+function count_lines_before_class_keyword( $file ) {
+	// Find the line that the `class` keyword occurs on so that we can use it to calculate an offset from the header.
+	$content = file_get_contents( $file ); // phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
+
+	// Find the class keyword and capture the number of characters in the string before this point.
+	if ( 0 === preg_match_all( '/^class /m', $content, $matches, PREG_OFFSET_CAPTURE ) ) {
+		return null;
+	}
+
+	// Support both styles of line endings.
+	$newlines = substr_count( $content, "\r\n", 0, $matches[0][0][1] );
+	if ( $newlines > 0 ) {
+		return $newlines + 1;
+	}
+	$newlines = substr_count( $content, "\n", 0, $matches[0][0][1] );
+	if ( $newlines > 0 ) {
+		return $newlines + 1;
+	}
+
+	return null;
+}
+
+/**
+ * Creates a map for converting file paths to src paths.
+ *
+ * @param string[] $report_file_paths An array containing all of the paths for the report.
+ * @return array A map describing how to transform built files into src coverage.
+ */
+function get_path_transformation_map( $report_file_paths ) {
+	// We're going to create a map describing how to transform files to src files.
+	// We're also going to store any metadata needed to perform the merge safetly.
+	$transformation_map = array();
+
+	// Scan the src directory so that we can create the map to convert between files.
+	$raw_src_files = scandir( ROOT_DIR . DIRECTORY_SEPARATOR . 'src' );
+	$src_file_map  = array();
+	foreach ( $raw_src_files as $file ) {
+		// Only PHP files will be copied.
+		if ( substr( $file, -4 ) !== '.php' ) {
+			continue;
+		}
+
+		$file = ROOT_DIR . DIRECTORY_SEPARATOR . 'src' . DIRECTORY_SEPARATOR . $file;
+		if ( ! file_exists( $file ) ) {
+			continue;
+		}
+
+		// We need to use the class keywork to address the line offset from injecting the header.
+		$class_line = count_lines_before_class_keyword( $file );
+		if ( ! isset( $class_line ) ) {
```

Comment:
Feel free to ignore, but unnecessary use of isset is a pet peeve of mine. p1614091140044600-slack-C6JKZ8MLK
```suggestion
		if ( null === $class_line ) {
```

Reply:
I'd never really thought deeply about the fact that `isset()` suppresses typos but I think that's a really good argument against using it unnecessarily.